plugin/kubernetes/node.go

```diff
@@ -0,0 +1,47 @@
+package kubernetes
+
+import (
+	"github.com/coredns/coredns/plugin/etcd/msg"
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+	"strings"
+)
+
+func (k *Kubernetes) Node(state request.Request) ([]msg.Service, int) {
+	node := state.Name()[:len(state.Name())-1]
+	nodeList := k.APIConn.NodeIndex(node)
+	services := []msg.Service{}
+	zonePath := msg.Path(state.Zone, coredns)
```

Comment:
IIRC, state.Zone contains the zone of the serverblock, not the zone that matched in the plugin zone declaration.

e.g. for ...

```
. {
  k8s_node myzone.
}
```

I think state.Zone is always ".", not "myzone.".